Czbank 34 fe registrace po ukonceni procesu registrace neni informace o tom zda registrace probehla uspesne

LucieKopasova opened the pull request.

LucieKopasova committed:
feat: add registration success page

LucieKopasova committed:
feat: add style and background

LucieKopasova committed:
style:add dark mode

vojtech-cerveny commented on src/app/register/success/page.tsx:
```diff
@@ -0,0 +1,74 @@
+
+  return (
+    <div>
+      <style jsx>{`
```
we don't use style. We use tailwindcss, can you please check it - it should also have support for everything what you did. Try some AI to transform it to tailwindcss

LucieKopasova replied:
I deleted the <style> block

postulkam replied:
@LucieKopasova FIY - you don't have to reply to every comment in your PR, you can add a like to it or click "Resolve conversation" (button below). With that, we will know that you did the change requested. :)

I usually add comments only if I want to discuss something. :)

vojtech-cerveny commented on src/app/register/success/page.tsx:
```diff
@@ -0,0 +1,74 @@
+          background-size: auto;
+        }
+      `}</style>
+      <div className={isDark ? "bg-money dark-bg-money" : "bg-money light-bg-money"}></div>
```
I would keep it simple - no image on background, keep it simple :)

LucieKopasova replied:
I deleted the background image.

vojtech-cerveny commented on src/app/register/success/page.tsx:
```diff
@@ -0,0 +1,74 @@
+        }
+      `}</style>
+      <div className={isDark ? "bg-money dark-bg-money" : "bg-money light-bg-money"}></div>
+      <div className="flex min-h-screen w-full items-center justify-center">
```
Here, I would love to see Card from components - it should have all styling what we need. Because right now, you are creating new styling, and this is not what we want. 
Check `/src/components/ui/card.tsx` - We have multiple usage of it, so it would be great to have consistency here.

LucieKopasova replied:
now I use  card components

vojtech-cerveny commented on src/app/register/success/page.tsx:
```diff
@@ -0,0 +1,74 @@
+          <h1 className="mb-2 text-2xl font-bold">Registration Successful!</h1>
+          <p className="mb-6 text-gray-600">You can now log in and start using Czechitoken.</p>
+          <a href="/profile">
+            <button className="w-full rounded-xl bg-[#ff007B] p-2 font-bold text-white hover:bg-[#d60068]">
```
Use component `Button` what we already have please

LucieKopasova replied:
now I use  Button components

vojtech-cerveny commented on src/app/register/success/page.tsx:
```diff
@@ -0,0 +1,74 @@
+    return () => {
+      document.removeEventListener("click", checkTheme);
+    };
+  }, []);
```
You don't need to check it. If you use our standard component, all is already setup to this. Or you can use tailwind pseudo class `dark:` 
For example for `bg` you already use this: `bg-white dark:bg-gray-900` so tailwind knows what color should use, if web is in darkmode.

LucieKopasova replied:
i deleted this block and am using components instead

vojtech-cerveny commented on src/app/register/success/page.tsx:
```diff
@@ -0,0 +1,74 @@
+          </div>
+          <h1 className="mb-2 text-2xl font-bold">Registration Successful!</h1>
+          <p className="mb-6 text-gray-600">You can now log in and start using Czechitoken.</p>
+          <a href="/profile">
```
Why this is redirecting to profile?

postulkam replied:
Better to redirect to homepage :)

<img width="1508" height="673" alt="image" src="https://github.com/user-attachments/assets/eddcc9d4-f7b0-4721-b92f-1c046b0830cf" />

LucieKopasova replied:
now it is  redirect to homepage

vojtech-cerveny commented on src/app/register/success/page.tsx:
```diff
@@ -0,0 +1,74 @@
+"use client";
+import { useEffect, useState } from "react";
+
+export default function RegisterSuccessPage() {
```
Do we need check if user is logged or not?

LucieKopasova replied:
Sorry, i am not sure what you mean, add some control for logged?

vojtech-cerveny replied:
```typescript
  const session = await userService.server.getSession(await headers());

  if (!session) {
    redirect("/signin");
  }
```

To check if user has session (is logged). But it would need some rewriting (this is server side function). 

Or @postulkam , wdyt?

postulkam replied:
At first, I agreed. But on second thought, I don’t think it’s necessary.

I checked where this component is initiated. It is inside the `onSuccess` of `await userService.client.signUp` function. Better-auth should handle the auto-sing-in after sign-up itself and be responsible for the session in this case.

LucieKopasova replied:
Should I do something there, or is it fine as it is?

vojtech-cerveny reviewed (changes requested):
https://www.better-auth.com/docs/basic-usage

Please check documenation - right now behaviour is that user is automatically signed in.

```
By default, the users are automatically signed in after they successfully sign up. To disable this behavior you can set autoSignIn to false.
```

So we can have "You are registered & Continue to app" or something like that. Or we can check autoSignIn to false and have your page. But it needs some love :)

But for KUDOS for this big PR - it is good start! ❤️

postulkam commented:
> https://www.better-auth.com/docs/basic-usage
> 
> Please check documenation - right now behaviour is that user is automatically signed in.
> 
> ```
> By default, the users are automatically signed in after they successfully sign up. To disable this behavior you can set autoSignIn to false.
> ```
> 
> So we can have "You are registered & Continue to app" or something like that. Or we can check autoSignIn to false and have your page. But it needs some love :)
> 
> But for KUDOS for this big PR - it is good start! ❤️


What about this - keep this page, but have "Continue to the app" in the big button in center?
(Btw, please, remove the message in the right corner, it is duplication of the information on page ;) )

<img width="1508" height="673" alt="image" src="https://github.com/user-attachments/assets/e7437757-da06-4253-a1e7-28143cde5ea5" />


Look at how the components are used, we are using shadcn/ui component library and it is really interesting. With the component libraries, you don't need to write everything from scratch and you can use their predefined components - https://ui.shadcn.com/ 
Good example is in `src/app/profile/page.client.tsx`, there is Card and Button component inside - these you can use in your solution. Inspire yourself and use it, then this PR will be perfect! :) 

P.S. Don’t be shy about contacting me on Discord in DM if you need any help. 🙂

LucieKopasova committed:
I modified it based on the prepared components.

LucieKopasova committed:
style: change style of button

LucieKopasova committed:
feat: Remove success toast, change button label

vojtech-cerveny commented on src/app/register/success/page.tsx:
```diff
@@ -27,10 +27,12 @@ export default function RegisterSuccessPage() {
           </CardContent>
           <CardFooter>
-            <Button className="w-full">Log in</Button>
+            <a href="/" className="w-full">
```
Please use `<Link></Link>` from  `import Link from "next/link";` instead of `<a></a>

postulkam commented on public/bg-money.png:
You can delete this image.

postulkam commented:
Nice work with the components! :)

One last thing. When you see the registration page, the card pushes the footer down and user have to scroll.
<img width="1510" height="822" alt="image" src="https://github.com/user-attachments/assets/15b7c913-d8b2-4a39-81a4-4bce95bb0073" />

But we need this, no footer out of the sight.
<img width="1510" height="822" alt="image" src="https://github.com/user-attachments/assets/f4683efc-3adf-4b71-b1e9-ffa067eaca40" />

Hint, try to copy the HTML code into ChatGPT with this question:
"This element (React app with Tailwind and Shadcn) is in the middle of the screen, but we have some footer there too and it pushes the footer and we have to scroll, how to put it in the middle without scrolling?"

LucieKopasova committed:
style: footer visible without scrolling

postulkam reviewed (dismissed):
LGTM

LucieKopasova committed:
merge: resolve conflict with develop in register form

Merge branch 'develop' into CZBANK-34-fe-registrace-po-ukonceni-procesu-registrace-neni-informace-o-tom-zda-registrace-probehla-uspesne

vojtech-cerveny reviewed (approved):
![](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExczN1dXR2OWF1ZzJuYjdqZWo1YXk0cGVrNWlocTRmYmwydmpneHYxbSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/NEvPzZ8bd1V4Y/giphy.gif)

LGTM!

please, when you will merge it, please add name of commit as 
`feature: [CZBANK-34] {name of this PR}`
